lib/src/facts/openbsd/memory_resolver.cc

```diff
@@ -0,0 +1,44 @@
+#include <internal/facts/openbsd/memory_resolver.hpp>
+#include <facter/execution/execution.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <sys/types.h>
+#include <sys/param.h>
+#include <sys/mount.h>
+#include <sys/sysctl.h>
+#include <sys/swap.h>
+#include <unistd.h>
+
+using namespace std;
+using namespace facter::execution;
+using namespace facter::util;
+
+namespace facter { namespace facts { namespace openbsd {
+
+    memory_resolver::data memory_resolver::collect_data(collection& facts)
+    {
+        data result;
+        size_t len;
```

Comment:
Can we move this declaration to its assignment?  I prefer not to declare variables apart from where they are initialized unless I have to.